lib/foreman/renderer/scope/variables/base.rb

```diff
@@ -25,8 +25,8 @@ def disk
           end
 
           def load_variables_base
-            @medium_provider = Foreman::Plugin.medium_providers.find_provider(host) if medium
-            if operatingsystem.respond_to?(:pxe_type)
+            @medium_provider = Foreman::Plugin.medium_providers.find_provider(host) if host.respond_to?(:medium) && medium
+            if operatingsystem && operatingsystem.respond_to?(:pxe_type)
```

Comment:
This does not solve the missing `#operatingsystem` method. we should probably add something like:

``` ruby
if host.respond_to?(:operatingsystem) && operatingsystem.respond_to?(:pxe_type)
```